faq.md

```diff
@@ -575,6 +575,144 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
 profile or release mode using the menu entries **Run > Flutter Run in Profile Mode**
 or **Release Mode**.
 
+### What programming paradigm does Flutter's framework use?
+
+Flutter is a multi-paradigm programming environment. Many programming techniques
+developed over the past few decades are used in Flutter. We use each one where
+we believe the strengths of the technique make it particularly well-suited. In
+no particular order:
+
+* Composition: The primary paradigm used by Flutter is that of using
+small objects with narrow scopes of behavior, composed together to
+obtain more complicated effects. Most widgets in the Flutter widget
+library are built in this way. For example, the Material
+[FlatButton](https://docs.flutter.io/flutter/material/FlatButton-class.html)
+class is built using a
+[MaterialButton](https://docs.flutter.io/flutter/material/MaterialButton-class.html)
+class, which itself is built using
+an [IconTheme](https://docs.flutter.io/flutter/widgets/IconTheme-class.html),
+an [InkWell](https://docs.flutter.io/flutter/material/InkWell-class.html),
+a [Padding](https://docs.flutter.io/flutter/widgets/Padding-class.html),
+a [Center](https://docs.flutter.io/flutter/widgets/Center-class.html),
+a [Material](https://docs.flutter.io/flutter/material/Material-class.html),
+an [AnimatedDefaultTextStyle](https://docs.flutter.io/flutter/widgets/AnimatedDefaultTextStyle-class.html),
+and a [ConstrainedBox](https://docs.flutter.io/flutter/widgets/ConstrainedBox-class.html).
+The [InkWell](https://docs.flutter.io/flutter/material/InkWell-class.html)
+is built using a [GestureDetector](https://docs.flutter.io/flutter/widgets/GestureDetector-class.html).
+The [Material](https://docs.flutter.io/flutter/material/Material-class.html)
+is built using an [AnimatedDefaultTextStyle](https://docs.flutter.io/flutter/widgets/AnimatedDefaultTextStyle-class.html),
+a [NotificationListener](https://docs.flutter.io/flutter/widgets/NotificationListener-class.html),
+and an [AnimatedPhysicalModel](https://docs.flutter.io/flutter/widgets/AnimatedPhysicalModel-class.html).
+And so on. It's widgets all the way down.
+
```

Comment:
When you say "Entire applications can be built with only StatelessWidget" you should probably mention the limitations. Like "Entire applications, such as those with no user input widgets, can be built with only StatelessWidget"

Reply:
Actually the very first application to be written with the runApp API was entirely stateless but still had user input (it just called runApp again when you provided input :-) ).

Added "`(Such applications can't easily have state, so are typically non-interactive.)`".